new.go

```diff
@@ -28,7 +29,20 @@ func newBuilder(store storage.Store, options BuilderOptions) (*Builder, error) {
 		name = options.Container
 	} else {
 		if image != "" {
-			name = image + "-" + name
+			prefix := image
+			s := strings.Split(prefix, "/")
+			if len(s) > 0 {
+				prefix = s[len(s)-1]
+			}
+			s = strings.Split(prefix, ":")
+			if len(s) > 0 {
+				prefix = s[0]
+			}
+			s = strings.Split(prefix, "@")
+			if len(s) > 0 {
+				prefix = s[0]
+			}
+			name = prefix + "-" + name
 		}
 	}
 	if name != "" {
```

Comment:
Is working-container a possibly valid value for a container name?  Do we want to error out in an else here if it is still working-container and/or set name initially to "" at line 23 and error out if there's no name?  I'm not sure, putting it up for consideration.